### PR TITLE
2D crosswell tomography app

### PR DESCRIPTION
**Pull request by sgkang**
Generate 2D tomographic inversion app

### CODE REVIEW

**Comment by fourndo** [comment]
Can we merge these two. We be cool to see the data change in real time. I get that the initial run will take a bit longer to generate the kernels, but after that would be better for the user not to have to run multiple cells.

image

**Comment by fourndo** [comment]
We can vote on this, but I would propose we abandon once and for all this whole p-q business and just call it
```p_s, p_x, p_y, p_z```
 instead. Peeps get really confused with p vs q, when they are already used to the s,x,y,z to denote the different regs.

image

**Comment by fourndo** [comment]
I am so puzzled by this. Somehow beta never gets updated, but phid always manage to stay on target...?
Will have to review master...

image

**Comment by lheagy** [comment]
to get the formatting test to pass, you  need to run `make format` from the base directory (this runs `black` to format the code
https://github.com/geoscixyz/geosci-labs/blob/43d1d2404d1c108f925e2d838cce1455dc539d8b/Makefile#L32-L33

Then running `make check` will run the linter and print out any issues that need to be addressed 
https://github.com/geoscixyz/geosci-labs/blob/43d1d2404d1c108f925e2d838cce1455dc539d8b/Makefile#L35-L37

and then commit any changes that were made (by black or by you) 

(this is exactly what is run when the style test runs on travis) 
https://github.com/geoscixyz/geosci-labs/blob/43d1d2404d1c108f925e2d838cce1455dc539d8b/.travis.yml#L29-L30

**Comment by lheagy** [comment]
you will need to have `black` and `pylint` installed (both are on pypi)

File: environment.yml [hunk 1]
```diff
@@ -17,4 +17,4 @@ dependencies:
     - deepdish
     - SimPEG
     - empymod
-    - git+https://github.com/geoscixyz/geosci-labs
+    - git+https://github.com/geoscixyz/geosci-labs.git@tomography-inversion
```
Comment (lheagy) [inline environment.yml]: @sgkang: would you mind reverting this? when we merge we will delete the branch, and this would break the binder

**Comment by sgkang** [comment]
Thanks @lheagy, I reverted environment.yml back to original. Will be great to merge this branch, so that we can distribute to participants at LASPIS.

File: environment.yml [hunk 2]
```diff
@@ -17,4 +17,4 @@ dependencies:
     - deepdish
     - SimPEG
     - empymod
-    - git+https://github.com/geoscixyz/geosci-labs
+    - git+https://github.com/geoscixyz/geosci-labs.git
```
Comment (lheagy) [inline environment.yml]: the `.git` isn't needed here. please see the original and then this is good to go!

Comment (lheagy) [inline environment.yml]:
```suggestion
    - git+https://github.com/geoscixyz/geosci-labs
```

**Pull request merged**